File: ios/Impermanence/Impermanence/Models/BellCatalog.swift
import Foundation

struct BellSound: Identifiable, Hashable {
    let id: Int
    let name: String
    let resourcePrefix: String

    var displayName: String { name }
    var resourceCandidates: [String] {
        ["\(resourcePrefix)_\(id)", "\(resourcePrefix)-\(id)", "\(resourcePrefix)\(id)", resourcePrefix]
    }
}

enum BellCatalog {
    static let sounds: [BellSound] = [
        BellSound(id: 0, name: "Classic Bell", resourcePrefix: "bell")
    ]

    static var defaultSound: BellSound { sounds.first! }

    static func sound(for id: Int) -> BellSound {
        sounds.first { $0.id == id } ?? defaultSound
    }
}

File: android/app/src/main/java/com/impermanence/impermanence/model/BellCatalog.kt
package com.impermanence.impermanence.model

object BellCatalog {
    private val sounds: List<BellSound> = listOf(
        BellSound(id = 0, name = "Classic Bell", resourcePrefix = "bell")
    )

    val defaultSound: BellSound
        get() = sounds.first()

    fun soundFor(id: Int): BellSound = sounds.firstOrNull { it.id == id } ?: defaultSound

    val all: List<BellSound>
        get() = sounds
}
